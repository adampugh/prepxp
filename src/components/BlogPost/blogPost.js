import React from "react";

import Navbar from "../UI/navbarLoggedIn";
import BlogPostContent from "./blogPostContent";
import Footer from "../UI/footer";

const blogPost = (props) => (
    <div>
        <Navbar />
        <BlogPostContent post={props.location.state.post} />
        <Footer />
    </div>
);

export default blogPost;