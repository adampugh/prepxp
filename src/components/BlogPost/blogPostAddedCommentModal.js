import React from "react";
import Modal from "react-modal";

const customStyles = {
    overlay : {
        zIndex: 2000,
        background: "rgba(0, 0, 0, 0.6)"
      }
}

const startSavedModal = (props) => (
    <Modal
        isOpen={props.isOpen}
        appElement={document.getElementById('root')} 
        onRequestClose={props.closeModal}
        style={customStyles}
        closeTimeoutMS={200}
        className="modal"
    >
        <div className="modal--delete">
            <span onClick={props.closeModal}><i className="fas fa-times"></i></span>
            <h2>Added comment</h2>
            <p className="modal__icon"><i className="fas fa-check"></i></p>
        </div>
    </Modal>
);

export default startSavedModal;