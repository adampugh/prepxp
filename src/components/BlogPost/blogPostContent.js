import React, { Component } from "react";
import { Link } from "react-router-dom";

class blogPostContent extends Component {
    componentDidMount() {
        window.scrollTo(0, 0);
        let el = document.getElementById("blogPost__content__text");
        if (el) {
            el.innerHTML = this.props.post.text;
        }
    }

    handleStartAddComment = () => {
        const text = document.querySelector("textarea").value;
        if (text.length > 0) {
            this.props.startAddComment(text, this.props.fbid, this.props.name);
            document.querySelector("textarea").value = "";
            this.props.openModal();
        }
    }

    handleStartDeleteComment = (postId) => {
        this.props.startDeleteComment(this.props.fbid, postId)
    }

    render() {
        return (
            <div className="blogPost__content">
                
                <div className="container-l">
                    <div className="blogPost__backLink">
                        <Link to="/prexp/blog"><i className="fas fa-arrow-left"></i> <p>Back to posts</p></Link>
                    </div>
                    <div className="blogPost__content__panel">
                        <h1>{this.props.post.title}</h1>
                        <h2 className="blogPost__content__byline">{this.props.post.byline}</h2>
                        <img src={this.props.post.img} alt="blog graphic" />
                        <div id="blogPost__content__text"></div>
                    </div>
                    <div className="blogPost__content__commentBox">
                        <h1>Comments Box</h1>
                        <textarea rows="6" placeholder="Leave a comment..."/>
                        <div className="blogPost__content__commentBox__button">
                            <button 
                                onClick={this.handleStartAddComment}
                                className="btn">Send</button>
                        </div>
                    </div>
                    <div className="blogPost__content__comments">
                        <h1>Comments</h1>
                        {
                            this.props.comments.length > 0 
                                ? this.props.comments.map((comment) => (
                                    <div className="blogPost__content__comments__comment" key={comment.postId}>
                                        <h3>{comment.name}</h3>
                                        <div className="blogPost__content__comments__comment__bottomDiv">
                                        <p>{comment.text}</p>
                                        {comment.uid === this.props.uid
                                            ? <span onClick={() => this.handleStartDeleteComment(comment.postId)}>
                                                <i className="fas fa-trash-alt"></i>
                                            </span>
                                            : null
                                        }
                                        </div>
                                    </div>
                                ))
                                : "No comments yet"
                        }
                    </div>

                </div>
            </div>
        );
    }
}

export default blogPostContent;