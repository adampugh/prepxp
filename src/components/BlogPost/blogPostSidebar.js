import React from "react";
import { Link } from "react-router-dom";

const blogPostSidebar = (props) => (
    <div className="blogPost__sidebar__container">
        
        <div className="blogPost__sidebar">
            <h1 className="blogPost__sidebar__title">Latest articles</h1>
            {
                props.posts.map(post => (
                    <Link key={post._id} to={{pathname: `/blog/${post.title.split(" ").join("-")}`, state:{post, posts: props.posts}}}>
                        <div className="blogPost__sidebar__item">
                            <h2>{post.title}</h2>
                            <p>{post.author}</p>
                            <hr />
                        </div>
                    </Link>
                ))
            }
        </div>
    </div>
);

export default blogPostSidebar;


// <img src={post.img} alt="post" />