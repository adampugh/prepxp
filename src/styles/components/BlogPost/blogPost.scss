.blogPost__grid {
    display: grid;
}

@media (min-width: $desktop-breakpoint) {
    .blogPost__container {
        background: $very-light-grey;
    }

    .blogPost__grid {
        grid-template-columns: 1fr 3fr;
    }
}