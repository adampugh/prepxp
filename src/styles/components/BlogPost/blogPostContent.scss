.blogPost__content {
    padding: $padding-s 0;
    background: $very-light-grey;
    h1 {
        @include header;
    }
    img {
        padding: $padding-s 0;
        width: 100%;
    }
    a {
        color: $off-purple;
        font-weight: bolder;
    }
    span {
        font-style: italic;
    }
}

.blogPost__backLink {
    padding-bottom: $padding-s;
    p {
        display: inline;
    }
}

.blogPost__content__byline {
    color: $light-grey;
    font-weight: 400;
}

.blogPost__content__text {
    font-size: $font-s;
}

.blogPost__content__panel {
    background: white;
    box-sizing: border-box;
    padding: $padding-s;
    @include boxshadow;
}


@media (min-width: $desktop-breakpoint) {
    .blogPost__content__panel {
        padding: $padding-m;
    }
}