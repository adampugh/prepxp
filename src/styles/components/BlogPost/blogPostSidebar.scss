.blogPost__sidebar__container {
    display: none;
}


@media (min-width: $desktop-breakpoint) {
    .blogPost__sidebar__title {
        @include header;
        &::after {
            content: "";
            display: block;
            width: 50px;
            height: 0.125rem;
            margin: 1rem 0;
            background: $off-purple;
        }
    }

    .blogPost__sidebar__container {
        display: block;
        // background: $very-light-grey;
        background: $light-purple;
    }
    
    .blogPost__sidebar {
        // height: 100%;
        // border: 1px solid $light-grey;
        margin: 80px 0 20px $padding-s;
        background: white;
        padding: 20px;
        @include boxshadow;
    }

    .blogPost__sidebar__item {
        // img {
        //     width: 100%;
        // }
        hr {
            border: 1px solid $very-light-grey;
            width: 100%;
        }
        p {
            color: $off-purple;
        }
        &:hover {
            h2 {
                color: $off-purple;
            }
        }
    }
}