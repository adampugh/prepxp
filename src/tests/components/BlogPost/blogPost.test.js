import React from "react";
import { shallow } from "enzyme";

import BlogPost from "../../../components/BlogPost/blogPost";
import posts from "../../fixtures/blogPosts";

describe("<BlogPost />", () => {
    it("should match snapshot", () => {
        let locationObj = {
            state: {
                post: posts[0]
            }
        }
        const wrapper = shallow(<BlogPost location={locationObj} />);
        expect(wrapper).toMatchSnapshot();
    });
})