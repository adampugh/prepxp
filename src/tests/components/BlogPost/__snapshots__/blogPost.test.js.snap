// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogPost /> should match snapshot 1`] = `
<div>
  <startSavedModal
    closeModal={[Function]}
    isOpen={false}
  />
  <Connect(Navbar) />
  <div
    className="blogPost__container"
  >
    <div
      className="container"
    >
      <div
        className="blogPost__grid"
      >
        <blogPostSidebar />
        <blogPostContent
          closeModal={[Function]}
          comments={Array []}
          fbid="134wegd"
          openModal={[Function]}
          post={
            Object {
              "byline": "yesyesyeys",
              "id": "1234",
              "img": "http://via.placeholder.com/300x300",
              "text": "sf sdfg dfasd df dg asdf sdf adsf sadf yea oia the ruler is bakc ofi course is i sfsf aoosd sgodf sfoeg wefodfguonfls sdfsd fsd sgosfsd sdfpsdf sdgxfogfdng  sdfosf sgosf sdfosg sdosf sgosdf sdfosdg sfosdf sd",
              "title": "5 Tips to get things done",
            }
          }
        />
      </div>
    </div>
  </div>
  <footer />
</div>
`;
